src/statscollector/StatsCollector.ts

```diff
@@ -1,63 +1,446 @@
+    const event = 'audio' + this.toSuffix(AudioLogEvent[eventName]);
+    this.logEvent(event, attributes);
+  }
+
+  logVideoEvent(eventName: VideoLogEvent, attributes?: { [id: string]: string }): void {
+    const event = 'video' + this.toSuffix(VideoLogEvent[eventName]);
+    this.logEvent(event, attributes);
+  }
+
+  private logEventTime(
+    eventName: string,
+    timeMs: number,
+    attributes: { [id: string]: string } = {}
+  ): void {
+    const finalAttributes = {
+      ...attributes,
+      call_id: this.audioVideoController.configuration.meetingId,
+      client_type: DefaultStatsCollector.CLIENT_TYPE,
+      metric_type: 'latency',
+    };
+    this.logger.debug(() => {
+      return `[DefaultStatsCollector] ${eventName}: ${JSON.stringify(finalAttributes)}`;
+    });
+    this.metricsAddTime(eventName, timeMs, finalAttributes);
+  }
+
+  logMeetingSessionStatus(status: MeetingSessionStatus): void {
+    // TODO: Generate the status event name given the status code.
+    const statusEventName = `${status.statusCode()}`;
+
+    this.logEvent(statusEventName);
+    const statusAttribute: { [id: string]: string } = {
+      status: statusEventName,
+      status_code: `${status.statusCode()}`,
+    };
+    this.logEvent('meeting_session_status', statusAttribute);
+    if (status.isTerminal()) {
+      this.logEvent('meeting_session_stopped', statusAttribute);
+    }
+    if (status.isAudioConnectionFailure()) {
+      this.logEvent('meeting_session_audio_failed', statusAttribute);
+    }
+    if (status.isFailure()) {
+      this.logEvent('meeting_session_failed', statusAttribute);
+    }
+  }
+
+  logLifecycleEvent(
+    lifecycleEvent: MeetingSessionLifecycleEvent,
+    condition: MeetingSessionLifecycleEventCondition
+  ): void {
+    const attributes: { [id: string]: string } = {
+      lifecycle_event: `lifecycle${this.toSuffix(MeetingSessionLifecycleEvent[lifecycleEvent])}`,
+      lifecycle_event_code: `${lifecycleEvent}`,
+      lifecycle_event_condition: `condition${this.toSuffix(
+        MeetingSessionLifecycleEventCondition[condition]
+      )}`,
+      lifecycle_event_condition_code: `${condition}`,
+    };
+    this.logEvent('meeting_session_lifecycle', attributes);
+  }
+
+  private logEvent(eventName: string, attributes: { [id: string]: string } = {}): void {
+    const finalAttributes = {
+      ...attributes,
+      call_id: this.audioVideoController.configuration.meetingId,
+      client_type: DefaultStatsCollector.CLIENT_TYPE,
+    };
+    this.logger.debug(() => {
+      return `[DefaultStatsCollector] ${eventName}: ${JSON.stringify(finalAttributes)}`;
+    });
+    this.metricsLogEvent(eventName, finalAttributes);
+  }
+
   /**
-   * Starts collecting statistics.
+   * WEBRTC METRICS COLLECTION.
    */
+
   start(
     signalingClient: SignalingClient,
     videoStreamIndex: VideoStreamIndex,
-    clientMetricReport?: ClientMetricReport
-  ): boolean;
+    clientMetricReport?: DefaultClientMetricReport
```

Comment:
This isn't used so maybe just remove it? Also could we do the same for ClientMetricReport (remove the interface and just keep the class) ?

Reply:
>This isn't used so maybe just remove it?

```js
if (clientMetricReport) {
  this.clientMetricReport = clientMetricReport;
} else {
  this.clientMetricReport = new DefaultClientMetricReport(
    this.logger,
    this.videoStreamIndex,
    this.audioVideoController.configuration.credentials.attendeeId
  );
}
```
Are you talking about that we have no actual use cases to use `clietnMetricReport` parameter?

>Also could we do the same for ClientMetricReport (remove the interface and just keep the class) ?

Sure, I can verify if we are able to do that and if so I will raise another PR for it. I prefer not to put them in a singe PR.

Reply:
Yeah we don't. Already check :). It is related to this so might as well.

Reply:
Remove the parameter from API.
Remove `ClientMetricReport` interface.
Rename `DefaultClientMetricReport` to `ClientMetricReport`.

We need to export ClientMetricReport due to metricDidReceive(clientMetricReport: ClientMetricReport).